### PR TITLE
[OHIF-251] - Double click to maximise viewport/ revert to previous layout.

### PR DESCRIPTION
**Pull request by JamesAPetts**
Includes cornerstoneTools 4.18.0 which allows us to cleanly cancel and broadcast removed toolState if we destroy the element whilst making a cornerstoneTools annotation.

Otherwise the viewport logic is common to all viewport types.

Needs to prevent click through on `ViewportActionbar` as right now jumping between measurements quickly triggers a double click on the cornerstone element.

### CODE REVIEW

File: extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js [hunk 1]
```diff
@@ -399,14 +399,24 @@ function OHIFCornerstoneSRViewport({
...
+  const label =
+    viewports.length > 1
+      ? _viewportLabels[firstViewportIndexWithMatchingDisplaySetUid]
+      : '';
+
   return (
     <>
       <ViewportActionBar
+        onDoubleClick={evt => {
+          debugger;
```
Comment (dannyrb) [inline extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js:411]:
```suggestion
```

File: extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js [hunk 2]
```diff
@@ -432,18 +441,23 @@ function OHIFCornerstoneSRViewport({
...
+          onElementEnabled={onElementEnabled}
+          viewportIndex={viewportIndex}
+          imageIds={imageIds}
+          imageIdIndex={currentImageIdIndex}
+          // TODO: ViewportGrid Context?
+          isActive={true} // todo
+          isStackPrefetchEnabled={true} // todo
+          isPlaying={false}
+          frameRate={24}
+          isOverlayVisible={false}
```
Comment (dannyrb) [inline extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js:458]: https://precisionmetrics.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=OHIF&modal=detail&selectedIssue=OHIF-269

File: extensions/measurement-tracking/src/viewports/TrackedCornerstoneViewport.js [hunk 3]
```diff
@@ -249,6 +249,12 @@ function TrackedCornerstoneViewport({
   return (
     <>
       <ViewportActionBar
+        onDoubleClick={evt => {
+          console.log('HI DANNY');
```
Comment (dannyrb) [inline extensions/measurement-tracking/src/viewports/TrackedCornerstoneViewport.js:253]:
```suggestion
```

File: extensions/measurement-tracking/src/viewports/TrackedCornerstoneViewport.js [hunk 4]
```diff
@@ -249,6 +249,12 @@ function TrackedCornerstoneViewport({
...
+          debugger;
```
Comment (dannyrb) [inline extensions/measurement-tracking/src/viewports/TrackedCornerstoneViewport.js:254]:
```suggestion
```

**Comment by dannyrb** [comment]
LGTM 👍

**Comment by richsmith123** [comment]
I don't see this feature has been implemented in the main branch as I have downloaded a copy last week.  Please correct me if I'm mistaken.

If this has not be merged to the main branch how do I go by making this change happen.  I see 18 files have been modified in this thread and some are outdated.  Any advice would be much appreciated!

Thanks,
Ric

**Pull request merged**